package/MDAnalysis/core/topologyobjects.py

```diff
@@ -655,7 +658,11 @@ def indices(self):
         --------
         to_indices : function that just returns `indices`
         """
-        return self._bix
+        if not self:
+            # empty TG
+            return np.zeros((0, 3), dtype=np.int32)
```

Comment:
Why `(0, 3)`? Shouldn't the number of columns depend on the topology attribute (*i.e.* 2 for bonds, 3 for anges, 4 for dihedrals...)?

Reply:
oops, yes